File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n1 = word1.size(), n2 = word2.size();
        int i = 0, j = 0;
        string res;
        while (i < n1 && j < n2) {
            res += word1[i++];
            res += word2[j++];
        }
        while (i < n1) {
            res += word1[i++];
        }
        while (j < n2) {
            res += word2[j++];
        }
        return res;
    }
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int m = word1.length(), n = word2.length();
        int i = 0, j = 0;
        String result = new String("");
        while (i < m && j < n) {
            result += word1.charAt(i);
            result += word2.charAt(j);
            i++;
            j++;
        }
        while (i < m) {
            result += word1.charAt(i);
            i++;
        }
        while (j < n) {
            result += word2.charAt(j);
            j++;
        }
        return result;
    }
}